ui-personalinfo/src/main/kotlin/edu/stanford/spezi/ui/personalinfo/UserProfile.kt

```diff
@@ -27,24 +27,46 @@ import androidx.compose.ui.unit.sp
 import edu.stanford.spezi.core.logging.SpeziLogger
 import edu.stanford.spezi.ui.Colors
 import edu.stanford.spezi.ui.ImageResource
-import edu.stanford.spezi.ui.ImageResourceComposable
 import edu.stanford.spezi.ui.SpeziTheme
 import edu.stanford.spezi.ui.StringResource
 import edu.stanford.spezi.ui.ThemePreviews
 import edu.stanford.spezi.ui.lighten
 import kotlin.math.min
 
+interface UserProfileImageLoader {
```

Comment:
Let's consider including [coil](https://github.com/coil-kt/coil) in a follow up. We can extend our Image resource with a case data class Remote(val url ...), and handle this internally to use coils [AsyncImage](https://github.com/coil-kt/coil/blob/main/coil-compose/src/commonMain/kotlin/coil3/compose/SingletonAsyncImage.kt#L46). It also supports api for fallbacks and place holders in case loading fails. And we dont need to create custom loaders for every use case then. For instance, we can apply it for Contact use case also

Reply:
Yes, we can provide an AsyncImageResource in the ui-package and make use of it here - I would actually keep it separate from ImageResource though to make a clear separation between cases where we allow for asynchronous loading and others where async loading would need to be handled separately and we actually expect an image to be available immediately.